File: IronPython/Excel/excel_to_csv.py
# coding: utf-8
#
# Excel の 1シート目を CSV 形式で出力するスクリプト
import sys

if len(sys.argv) < 3:
	print "ipy %s <excel file> <dest csv file>" % sys.argv[0]
	sys.exit()

from System import *
from System.IO import *

import clr
clr.AddReference("office")
clr.AddReference("Microsoft.Office.Interop.Excel")

from Microsoft.Office.Core import *
from Microsoft.Office.Interop.Excel import *

app = ApplicationClass()
book = app.Workbooks.Open(FileInfo(sys.argv[1]).FullName)

book.Sheets[1].SaveAs(FileInfo(sys.argv[2]).FullName, XlFileFormat.xlCSV)

book.Close(False)
app.Quit()

File: WSH/excel/excel_to_csv/excel_to_csv.js

var args = WScript.Arguments;

if (args.length < 1) {
	WScript.Echo("<excel file>");
	WScript.Quit();
}

var fs = WScript.createObject("Scripting.FileSystemObject");
var app = WScript.CreateObject("Excel.Application");

// 確認メッセージの非表示設定
app.DisplayAlerts = false;

var excelFile = args(0)
var csvFile = fs.GetBaseName(excelFile) + ".csv";

try {
	var book = app.Workbooks.Open(fs.GetAbsolutePathName(excelFile));

	// 1シート目を CSV ファイルへ保存
	book.Sheets(1).SaveAs(fs.GetAbsolutePathName(csvFile), 6);

	book.Close(false);
} catch(e) {
}

app.Quit();
app = null;
